RuminantFarmSystems/RuFaS
# Update Expiring Github Dependency

Updates an expiring Github dependency we're using as well as updating some of the soon-to-be-deprecated GH action commands and updating a package to update the README to the latest version.

### Context

Issue(s) closed by this pull request: closes #1453 closes #1657

Ed's edit: closes #832

- [N/A] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
      
How big are the changes in the PR? Would you nominate it for a major version upgrade?

### What

The jitterbug package we used in our github actions to collect all the changed files is expiring soon and currently is giving us warnings on some of the soon-to-be-deprecated Github commands it uses (`set-output`). This PR seeks to replace that with a longer-term solution of manually gathering those files.

The EndBug/add-and-commit package was also using the `set-output` GH command but they have a newly updated version (9) which I've updated us to use here and it fixed the warnings.

Also removes another use of `set-output` which was unneeded.

### Why

There have been warnings that these features/packages are being deprecated and fixing them before they are deprecated will help our GH actions continue to run as expected.

### How

Rather than switching to another dependency, the GH action is updated to now manually get the changed files (both Python files and non-Python files).

Removed other usages of `set-output` by updating dependencies to their latest versions or checking for the need for our own use of `set-output` and removing it or updating it where applicable.

### Test plan

To test:
- Changed a .py file and ensured modified files were being accurately found (they were).
- Added that changed .py file to ensure protected paths were still able to be found (they were).
- Check that other functionalities dependent on that list of modified files were still functioning as intended (Black, e.g.).
- Checked that README was still being properly updated.
